src/components/layout/header/scroll-effects.ts

```
@@ -3,58 +3,94 @@
  * Handles header scroll effects and backdrop blur
  */
 
-export class ScrollEffectsManager {
-  private headers: NodeListOf<HTMLElement> | null = null;
+import type { Destroyable, EventCleanup } from "./types";
+
+export class ScrollEffectsManager implements Destroyable {
+  private headers: HTMLElement[] = [];
   private scrollThreshold: number = 20;
```

Biome (1.9.4)

[error] This type annotation is trivially inferred from its initialization.

Safe fix: Remove the type annotation.


(lint/style/noInferrableTypes)